sdk/core/Azure.Core/samples/Serialization.md

```diff
@@ -0,0 +1,164 @@
+# Azure.Core public serialization samples
+
+## Using explicit cast
```

Comment:
It might be helpful to begin with a short summary of what the interfaces are and how they are intended to be used.  For Azure.Core, it can be helpful to think about this from the perspective of an end-user of a library, as well as a library author.  Should library authors implement IModelSerializable on all models?  Will the generator handle it for them?  When do they need to also implement IModelJsonSerializable?  What does an end-user need to know they can do with a model that implements these interfaces?  Should they understand ModelSerializer and how to use it?  When do I need to be thinking about ModelJsonConverter and why?  It'd be great to get all these questions answered at the start with some short illustrative samples before diving into further details and all the options types below.  That way, when I get to those sections, as a reader I would know what I'm looking at and why it's useful to me.

Reply:
moved comment to ReadMe PR: https://github.com/Azure/azure-sdk-for-net/pull/37998/files#r1282782409